### PR TITLE
ruby and rubygem-rexml: patch CVE-2024-49761

### PR DESCRIPTION
**Pull request by Redent0r**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
- ruby: patch CVE-2024-49761
- rubygem-rexml: patch CVE-2024-49761

###### Change Log  

- CVE-2024-49761

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2024-49761

###### Test Methodology

buddy build: 
- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=669847&view=results

### CODE REVIEW

**Comment by Redent0r** [comment]
Able to install both on 3.0 container
image

image

**Review by nicogbg — APPROVED** [review]

**Review by anphel31 — APPROVED** [review]

**Review by mbykhovtsev-ms — APPROVED** [review]

File: SPECS/ruby/ruby.spec [hunk 1]
```diff
@@ -408,6 +410,9 @@ sudo -u test make test TESTS="-v"
 %{_rpmconfigdir}/rubygems.con
 
 %changelog
+* Tue Nov 05 2024 Saul Paredes <saulparedes@microsoft.com> - 3.1.4-3
```
Comment (anphel31) [inline SPECS/ruby/ruby.spec:413]: I think this version should be 3.3.3-3

File: SPECS/rubygem-rexml/rubygem-rexml.spec [hunk 2]
```diff
@@ -3,13 +3,14 @@
...
 Version:        3.3.4
-Release:        1%{?dist}
+Release:        2%{?dist}
 License:        BSD
 Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
 Group:          Development/Languages
 URL:            https://github.com/ruby/rexml
 Source0:        https://github.com/ruby/rexml/archive/refs/tags/v%{version}.tar.gz#/%{gem_name}-%{version}.tar.gz
+Patch0: CVE-2024-49761.patch
```
Comment (anphel31) [inline SPECS/rubygem-rexml/rubygem-rexml.spec:13]: Can you fix the indentation here?

File: SPECS/rubygem-rexml/rubygem-rexml.spec [hunk 3]
```diff
@@ -34,6 +35,9 @@ gem install -V --local --force --install-dir %{buildroot}/%{gemdir} %{gem_name}-
 %{gemdir}
 
 %changelog
+* Tue Nov 05 2024 Saul Paredes <saulparedes@microsoft.com> - 3.2.7-2
```
Comment (anphel31) [inline SPECS/rubygem-rexml/rubygem-rexml.spec:38]: This version looks wrong - shouldn't it be 3.3.4-2?

**Comment by Redent0r** [comment]
Closing as I'm including this patch fix in https://github.com/microsoft/azurelinux/pull/10988 now to avoid conflicts

**Pull request closed without merging**